src/aws-cpp-sdk-core/include/aws/core/client/AWSClient.h

```diff
@@ -191,6 +191,8 @@ namespace Aws
             virtual void SetServiceClientName(const Aws::String& name);
 
             void AppendToUserAgent(const Aws::String& valueToAppend);
+
+            static bool DoesResponseGenerateError(const std::shared_ptr<Aws::Http::HttpResponse>& response);
```

Comment:
should not be public, could be protected so that child classes can use it, that would for for you in this case, but definitely should not be added to the public interface

Reply:
ok, it used to be not public, exposed it because it was a const function and I needed to check from the client if http response is an error. currently that check happens one level down which is not where I needed the check